### PR TITLE
841 breadcrumbs component

### PR DESCRIPTION
**Pull request by johannegg**
Made a breadcrumb component for the Control panel. The code is maybe a bit unpractical because of the many url cases.

### CODE REVIEW

File: frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { KEY, KeyValues } from '~/i18n/constants';
+
+export const BreadcrumbTitles: Record<string, KeyValues> = {
```
Comment (emilte) [inline frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts:3]: Dette trenger du ikke hardkode. Bare bruk ROUTES.frontend. Kan sette på type også

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    return (
+      <React.Fragment key={i}>
+        <span className={styles.separator}>&nbsp;&gt;&nbsp;</span>
+        <Link to={baseUrl + url} className={styles.link}>
+          {title}
+        </Link>
+      </React.Fragment>
+    );
+  });
```
Comment (emilte) [inline frontend/src/Components/Breadcrumb/Breadcrumb.tsx:64]: Vurder sub komponent

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const gangId = useParams().gangId;
+
+  useEffect(() => {
+    if (gangId) {
+      getGang(gangId)
+        .then((gang) => {
+          setGang(gang);
+        })
+        .catch((error) => console.error('Error fetching gang:', error));
+    }
```
Comment (emilte) [inline frontend/src/Components/Breadcrumb/Breadcrumb.tsx:26]: Anbefaler guard pattern (early return)
```suggestion
    if (!gangId) return
      getGang(gangId)
        .then((gang) => {
          setGang(gang);
        })
        .catch((error) => console.error('Error fetching gang:', error));
```

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { GangDto } from '~/dto';
+import { KEY } from '~/i18n/constants';
+import styles from './Breadcrumb.module.scss';
+import { BreadcrumbTitles } from './BreadcrumbTitles';
+
+export function Breadcrumb() {
+  const { t } = useTranslation();
+  const location = useLocation();
+  const segments = location.pathname.split('/').filter(Boolean);
+  const baseUrl = 'http://localhost:3000';
```
Comment (emilte) [inline frontend/src/Components/Breadcrumb/Breadcrumb.tsx:15]: for later

File: frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts (hunk 1 above)
Comment (johannegg) [reply frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts:3]: @emilte Lurer litt på hva du mener med at jeg ikke trenger å hardkode det. Tenkte dette var en grei måte å hente ut titlene som skal stå i breadcrumb-en på.

Comment (emilte) [reply frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts:3]: Når rutene i frontend byttes brekker disse. Dataene er nå duplisert. Hvis det skal være likt, vurder ROUTES med en metode som stripper `:params`

**Comment by robines** [comment]
Reworked this PR. Notable changes:

* Switch Samf4 to use a data router, allowing us to use the react router data API. [More info](https://reactrouter.com/en/6.23.0/routers/picking-a-router)
* Vastly simplify the Breadcrumb component using the now-available `useMatches` hook from the react router data API.
* We can now add crumbs to each route in `AppRoutes.tsx` using `handle`

TODO:

* Get recruitment, gang, and position data using route loaders. This seems worthy of its own PR/issue. For now, the IDs are used, which is fine.
* Consider cleaning up `AppRoutes.tsx`

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx (hunk 4 above)
Comment (emilte) [reply frontend/src/Components/Breadcrumb/Breadcrumb.tsx]: notis til meg selv bare, stusser litt over at den er her

File: frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts (hunk 1 above)
Comment (robines) [reply frontend/src/Components/Breadcrumb/BreadcrumbTitles.ts]: Utdatert, resolve?

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (robines) [reply frontend/src/Components/Breadcrumb/Breadcrumb.tsx]: Utdatert, resolve?

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx (hunk 3 above)
Comment (robines) [reply frontend/src/Components/Breadcrumb/Breadcrumb.tsx]: Utdatert, resolve?

File: frontend/src/Components/Breadcrumb/Breadcrumb.tsx (hunk 4 above)
Comment (robines) [reply frontend/src/Components/Breadcrumb/Breadcrumb.tsx]: Utdatert, resolve?

**Review by Snorre98 — COMMENTED** [review]
Fix routes for "create event" and for "create opptak"

**Review by Snorre98 — APPROVED** [review]
LGTM

**Pull request merged**